give_kudos.py

```diff
@@ -1,32 +1,31 @@
 import os
 import time
 
-from playwright.sync_api import sync_playwright, TimeoutError
+from playwright.sync_api import sync_playwright
 
 BASE_URL = "https://www.strava.com/"
 
 class KudosGiver:
     """
     Logins into Strava and gives kudos to all activities under
-    Following. Additionally, scrolls down to check for more activities
-    until no more kudos can be given at this time.
+    Following.
     """
-    def __init__(self, max_retry_scroll=10, max_run_duration=540) -> None:
+    def __init__(self, max_run_duration=540) -> None:
         self.EMAIL = os.environ.get('STRAVA_EMAIL')
         self.PASSWORD = os.environ.get('STRAVA_PASSWORD')
 
         if self.EMAIL is None or self.PASSWORD is None:
             raise Exception(f"Must set environ variables EMAIL AND PASSWORD. \
                 e.g. run export STRAVA_EMAIL=YOUR_EMAIL")
 
-        self.max_retry_scroll = max_retry_scroll
         self.max_run_duration = max_run_duration
+        self.start_time = time.time()
         self.kudos_button_pattern = '[data-testid="kudos_button"]'
+        self.web_feed_entry_pattern = '[data-testid=web-feed-entry]'

```

Comment:
This is related to the general approach: I wonder if we could skip web-feed-entry all together and use `kudos_comments_container`, since we are essentially finding the n_th `kudos_comments_container` anyway?
```python
kudos_container = web_feed.get_by_test_id("kudos_comments_container").nth(j)
```

Reply:
<img width="1055" alt="image" src="https://user-images.githubusercontent.com/12427722/233695079-d3d767dd-b046-49c7-bb6e-91cf445570d6.png">

You need the `web-feed-entry` to iterate over the GroupActivity entries and check if the entry is not from yourself via the `entry-header`. If not we can use the `kudos_comments_container`.

Reply:
Sounds good, thanks